### PR TITLE
[r3.5] cl, cmd/utils: derive the column retention window from the chain config

### PR DESCRIPTION
**Pull request by lystopad**
Cherry-pick of #23610 to `release/3.5`.

## r3.5-specific adaptations

`release/3.5` prunes columns by keep-distance (`peerDas.Prune(keepSlotDistance)`) rather than by
absolute floor (`PruneBelow`), so the original's `specColumnFloor` does not apply. The epoch
alignment lands instead as one extra epoch of slack in the derived distance:
`(MIN_EPOCHS + 1) * SLOTS_PER_EPOCH`.

That matters because the same distance sets `EarliestAvailableSlot` — the earliest slot we
advertise as servable — and `peerdas.Prune` computes it without the `subdivisionSlot` rounding
the on-disk path applies. An exact `MIN_EPOCHS * SLOTS_PER_EPOCH` distance therefore cuts above
the epoch boundary whenever the head sits inside an epoch, by up to `SLOTS_PER_EPOCH - 1` slots.
Note this is pre-existing on `release/3.5`, not introduced by defaulting the flag to `0`: the old
`131072` constant produced the same unaligned cut. Rounding up keeps the cut at or below the
boundary for any head position and needs no clock read in the stage.

Retention after this change: Gnosis and Chiado `4097 * 16 = 65552` slots, down from `131072`.
Mainnet `4097 * 32 = 131104`, i.e. one epoch more than before.

`cl/phase1/stages/cleanup_and_pruning_test.go` does not exist on this branch, so the original's
additions to it are replaced with a small file written against the keep-distance API, using
`memdb.NewTestDB` (this branch has no `mdbxtest`).

## Verification

Three new tests: the derived distance, an explicit slot count passing through unchanged, and the
boundary property checked across every head offset within an epoch. Mutation-verified — dropping
the `+1` fails the boundary test with `head 80001 cuts above the first slot the node must still
serve`, and restoring the `131072` flag default fails
`TestCaplinColumnKeepSlots_UnsetDefersToChainConfig`.

`cl/phase1/stages`, `cmd/utils` and `cl/clparams` pass; `make lint` clean.

The flag test is rewritten for `urfave/cli` v2 (`cli.NewApp()` / `*cli.Context`); `main` and
`release/3.6` are on v3. Resolving the pick also required dropping an unrelated
`TestNewP2PConfig_DiscoveryDefaults` that git pulled in from the surrounding hunk context —
verified absent from the result.

### CODE REVIEW

File: docs/site/docs/fundamentals/caplin.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ In addition, Caplin can backfill recent blobs for an op-node or other uses with
 
 For nodes participating in PeerDAS (EIP-7594), Caplin retains data column sidecars for a configurable window:
 
-* `--caplin.columns-keep-slots` (default: `131072`, ~18 days): Number of slots to retain PeerDAS data column sidecars. The default matches `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS × SLOTS_PER_EPOCH`. Increase this value for DA oracle or rollup nodes that require a longer column history.
+* `--caplin.columns-keep-slots` (default: `0`): Number of slots to retain PeerDAS data column sidecars. `0` uses the chain's own spec window, `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS × SLOTS_PER_EPOCH`, so the retained slot count and the wall-clock duration it covers both follow the chain (on Ethereum mainnet, 131072 slots, ~18 days). Increase this value for DA oracle or rollup nodes that require a longer column history.
```
Comment (domiwei) [inline docs/site/docs/fundamentals/caplin.md]: Non-blocking: this says the default resolves to MIN_EPOCHS times SLOTS_PER_EPOCH and gives 131072 for mainnet, but the implementation uses (MIN_EPOCHS + 1) times SLOTS_PER_EPOCH, which is 131104. Could we document the alignment safety margin and actual effective value so operator retention and disk expectations match runtime behavior?

File: cl/phase1/stages/cleanup_and_pruning.go [hunk 2]
```diff
@@ -28,9 +29,17 @@ func cleanupAndPruning(ctx context.Context, logger log.Logger, cfg *Cfg, args Ar
...
 	return nil
 }
+
+// specColumnKeepSlots is the retention distance for MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS.
+// The window is stated in epochs and measured from the start of current_epoch - MIN_EPOCHS, so an
+// exact epochs * SLOTS_PER_EPOCH distance cuts above that boundary whenever the head sits inside
+// an epoch. The extra epoch keeps the cut at or below it, which matters because this distance also
+// sets the earliest slot we advertise as servable.
+func specColumnKeepSlots(beaconCfg *clparams.BeaconChainConfig) uint64 {
+	return (beaconCfg.MinEpochsForDataColumnSidecarsRequests + 1) * beaconCfg.SlotsPerEpoch
```
Comment (domiwei) [inline cl/phase1/stages/cleanup_and_pruning.go]: This arithmetic is fed by uint64 fields from --caplin.custom-config without a range check. With MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS set to MaxUint64, the addition wraps to zero; PeerDas.Prune(0) can then remove all completed buckets before the current slot and advance earliest_available_slot. Because this PR makes the chain-derived path the default, please use checked or saturating addition and multiplication, conservatively retaining on overflow, and add boundary tests proving overflow cannot reach destructive pruning.

**Review by domiwei — CHANGES_REQUESTED** [review]
Requesting changes for the unchecked retention-window arithmetic. Values supplied through the custom chain config can wrap the derived keep distance to zero and reach destructive column pruning. Please make the addition and multiplication overflow-safe and add focused boundary coverage. The documentation mismatch noted inline is non-blocking.

**Review by domiwei — APPROVED** [review]
